packages/solana/src/__tests__/index-test.ts

```diff
@@ -50,6 +58,73 @@ describe("TurnkeySigner", () => {
     expect(transferTransaction.signatures.length).toBe(0);
     await signer.addSignature(transferTransaction, turnkeySolAddress);
     expect(transferTransaction.signatures.length).toBe(1);
+
+    const isValidSignature = nacl.sign.detached.verify(
+      transferTransaction.serializeMessage(),
+      transferTransaction.signature as Uint8Array,
+      bs58.decode(turnkeySolAddress)
+    );
+    expect(isValidSignature).toBeTruthy();
+  });
+
+  test("can sign a versioned Solana transfer against production", async () => {
+    if (!process.env.SOLANA_TEST_ORG_API_PRIVATE_KEY) {
+      // This test requires an env var to be set
+      throw new Error(
+        "This test requires SOLANA_TEST_ORG_API_PRIVATE_KEY to be set"
+      );
+    }
+
+    const client = new TurnkeyClient(
+      { baseUrl: "https://api.turnkey.com" },
+      new ApiKeyStamper({
+        apiPublicKey,
+        apiPrivateKey: process.env.SOLANA_TEST_ORG_API_PRIVATE_KEY,
+      })
+    );
+
+    const signer = new TurnkeySigner({
+      organizationId,
+      client,
+    });
+
+    const fromKey = new PublicKey(turnkeySolAddress);
+
+    const instructions = [
+      SystemProgram.transfer({
+        fromPubkey: fromKey,
+        // Destination doesn't matter, we set it to the Turnkey war chest!
+        toPubkey: new PublicKey("tkhqC9QX2gkqJtUFk2QKhBmQfFyyqZXSpr73VFRi35C"),
+        lamports: 10,
+      }),
+    ];
+
+    // create v0 compatible message
+    const messageV0 = new TransactionMessage({
+      payerKey: fromKey,
+      // Doesn't really matter since we're not going to broadcast this transaction!
+      recentBlockhash: "GZSq3KvoFVhE22CQsaWSAxBoAvRiZSs7xVNa6yPecMUu",
+      instructions,
+    }).compileToV0Message();
+
+    const transaction = new VersionedTransaction(messageV0);
+
+    // version transactions are initalized with a default signature
+    expect(transaction.signatures.length).toBe(1);
```

Comment:
really strange behavior, I wonder why that is 🤔 

(goes without saying: not your fault!)

Reply:
solana just likes to do things differently 🤷‍♂️